src/main/java/com/cburch/logisim/fpga/hdlgenerator/Hdl.java

```diff
@@ -83,6 +136,67 @@ public static String xorOperator() {
     return isVhdl() ? " XOR " : "^";
   }
 
+  public static String addOperator(String signalOne, String signalTwo, boolean signed) {
+    return (isVhdl() ? "std_logic_vector(" : "")
+            + typecast(signalOne, signed)
+            + " + "
+            + typecast(signalTwo, signed)
+            + (isVhdl() ? ")" : "");
+  }
+
+  public static String subOperator(String signalOne, String signalTwo, boolean signed) {
+    return (isVhdl() ? "std_logic_vector(" : "")
+            + typecast(signalOne, signed)
+            + " - "
+            + typecast(signalTwo, signed)
+            + (isVhdl() ? ")" : "");
+  }
+
+  public static String shiftlOperator(String signal, int width, int distance, boolean arithmetic) {
+    if (distance == 0) return signal;
+    return isVhdl() ? LineBuffer.format("{{1}}{{2}} & {{4}}{{3}}{{4}}", signal, splitVector(width - 1 - distance, 0), "0".repeat(distance), distance == 1 ? "'" : "\"")
+                    : LineBuffer.format("{{{1}}{{2}},{{{3}}{1'b0}}}", signal, splitVector(width - 1 - distance, 0), distance);
+  }
+
+  public static String shiftrOperator(String signal, int width, int distance, boolean arithmetic) {
+    if (distance == 0) return signal;
+    if (arithmetic) {
+      return isVhdl()
+        ? LineBuffer.format("({{1}} DOWNTO 0 => {{2}}({{1}})) & {{2}}{{3}}", width - 1, signal, splitVector(width - 1, width - distance))
+        : LineBuffer.format("{ {{{1}}{{{2}}[{{1}}-1]}},{{2}}{{3}}}", width, signal, splitVector(width - 1, width - distance));
+    } else {
+      return isVhdl()
+        ? LineBuffer.format("{{1}}{{2}}{{1}} & {{3}}{{4}", distance == 1 ? "'" : "\"", "0".repeat(distance), signal, splitVector(width - 1, width - distance))
+        : LineBuffer.format("{ {{{1}}{1'b0}},{{2}}{{3}}}", width, signal, splitVector(width - 1, width - distance));
```

Comment:
As you have space after first opening curly brace, shouldn't it be consistent to have space preceding last closing curly brace too?

Reply:
I only added this space because of the limitations of LineBuffer.format().
The space ideally shouldn't be there. IMO, adding another space that shouldn't be there does not improve the situation.